src/compiler/tsc.ts

```diff
@@ -285,6 +285,17 @@ namespace ts {
 
             setCachedProgram(compileResult.program);
             reportWatchDiagnostic(createCompilerDiagnostic(Diagnostics.Compilation_complete_Watching_for_file_changes));
+
+            const missingPaths = compileResult.program.getMissingFilePaths() || [];
+            missingPaths.forEach((path: Path): void => {
+                const fileWatcher = sys.watchFile(path, (_fileName: string, removed?: boolean) => {
+                    // removed = deleted ? true : (added ? false : undefined)

```

Comment:
This could use an enum rather than `true | false | undefined`. `true` -> `E.Deleted`, `false` -> `E.Added`, `undefined` -> `E.Changed`.